### PR TITLE
Revisit Session Management docs

### PR DESCRIPTION
**Pull request by marcusdacoregio**
Closes gh-12519

### CODE REVIEW

**Review by jzheaux — CHANGES_REQUESTED** [review]
Nice, @marcusdacoregio! I've left my feedback inline.

File: docs/modules/ROOT/pages/servlet/authentication/session-management.adoc [hunk 1]
```diff
@@ -16,53 +13,297 @@ This can be done by using the {security-api-url}org/springframework/security/web
...
+====
+
+<1> Add the `SecurityContextRepository` to the controller
+<2> Save the `SecurityContext` to the `SecurityContextRepository`
+
+And that's it.
+If you are not sure what `securityContextHolderStrategy` is in the above example, you can read more about it in the <<use-securitycontextholderstrategy, Using `SecurityContextStrategy` section>>.
+
+[[stateless-authentication]]
+=== Configuring persistence for Stateless Authentication
```
[the remark this answers is not in the document]
Comment (marcusdacoregio) [reply docs/modules/ROOT/pages/servlet/authentication/session-management.adoc:301]: I think I'll remove the `RequestCache` section and a ticket can be created to add some information about that.

**Comment by marcusdacoregio** [comment]
Thanks, @jzheaux, I've addressed your suggestions and updated the PR. I also reverted [this commit](https://github.com/spring-projects/spring-security/pull/12601/commits/19ccbbaec598c9d857440b7b4e54c84eab6eb673) based on @rwinch feedback.

**Review by jzheaux — APPROVED** [review]
Such a huge improvement, @marcusdacoregio. Thanks for your efforts.

**Pull request merged**